autoscale/src/main/java/com/sequenceiq/periscope/model/MonitoredStack.java

```diff
@@ -8,20 +8,31 @@
 
     private final ClusterManager clusterManager;
 
-    private final String stackCrn;
+    private String stackCrn;
+
+    private String stackName;
+
+    private String stackType;
 
     private final Long stackId;
 
     private final SecurityConfig securityConfig;
 
     private final Tunnel tunnel;
 
-    public MonitoredStack(ClusterManager clusterManager, String stackCrn, Long stackId, SecurityConfig securityConfig, Tunnel tunnel) {
+    public MonitoredStack(ClusterManager clusterManager, String stackName, String stackCrn,
+            String stackType, Long stackId, SecurityConfig securityConfig, Tunnel tunnel) {
         this.clusterManager = clusterManager;
-        this.stackCrn = stackCrn;
-        this.stackId = stackId;
         this.securityConfig = securityConfig;
         this.tunnel = tunnel;
+        this.stackId = stackId;
+        initStackContext(stackName, stackCrn, stackType);
+    }
+
+    private void initStackContext(String stackName, String stackCrn, String stackType) {
```

Comment:
Skip this? Set the fields in the constructor itself, so that they can be 'final'

Reply:
There is checkstyle limit on constructor to 5 executable statements hence had to refactor